pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -395,6 +395,17 @@ const updateTable: Record<number, (settings: any) => void> = {
       delete settings.extensions;
     }
   },
+  9: (_) => {
+    // On linux, this version deletes old instances of credential-server.json and rd-engine.json from paths.config
+    // These should always be in paths.appHome as they aren't config files
+    if (process.platform === 'linux') {
+      for (const filename of ['credential-server.json', 'rd-engine.json']) {
+        // Ignore nonexistent files (but if we're moving from settings 9 to 10 on linux, they will exist
+        // if the last session running settings version 9 wasn't followed by a factory-reset)
+        fs.rmSync(join(paths.config, filename), { force: true });
+      }
+    }
```

Comment:
Conceptually it seems just wrong to me that a schema migration can have side-effects that modify the filesystem.

A schema migration should just transform a JSON object into another JSON object and not have any observable side-effect.

E.g. the migration code should be called on `settings` API requests and deployment profiles as well (see #4939), and having those potentially touch the filesystem is just wrong.

Reply:
I finally reverted that code -- thought I did that last week.